sklearn/ensemble/tests/test_forest.py

```diff
@@ -181,6 +183,65 @@ def test_regression(name, criterion):
     check_regression_criterion(name, criterion)
 
 
+def test_poisson_vs_mse():
+    rng = np.random.RandomState(42)
+    n_train, n_test, n_features = 500, 500, 10
+    X = datasets.make_low_rank_matrix(n_samples=n_train + n_test,
+                                      n_features=n_features, random_state=rng)
+
+    coef = rng.uniform(low=-2, high=2, size=n_features) / np.max(X, axis=0)
```

Comment:
```suggestion
    X = datasets.make_low_rank_matrix(n_samples=n_train + n_test,
                                  n_features=n_features, random_state=rng)
    X = np.abs(X)
    X /= np.max(np.abs(X), axis=0)

    # We create a log-linear Poisson model
    coef = rng.uniform(low=-4, high=1, size=n_features)
```
This generates reasonable `coef`
```
array([-1.28649376, -0.00494814, -3.81872552,  0.92009075, -1.72546699,
       -2.2038443 , -2.53637781, -3.12821646, -0.04213046,  0.77707082])
```
and even better mean = lambda = exp(X @ coef):
```
np.min(np.exp(X @ coef)), np.max(np.exp(X @ coef)), np.mean(np.exp(X @ coef))
(0.00034419812245193335, 0.7339111016297659, 0.07910642768713219)
```
We prefer small lambda values because only then does Poisson deviate much from the Normal distribution.